client/src/GameState.ts

```diff
@@ -320,49 +310,44 @@ export class GameState {
     ): Map<number, number> {
         const linesOffset = new Map<number, number>();
         let offset = 0;
-        if (linesToClear.length > 0) {
-            task.range
-                .slice()
-                .reverse()
-                .forEach((lineIndex) => {
-                    if (lineIndex === linesToClear?.at(0)) {
-                        // this line is an empty, cleared line. all lines above should fall with one more offset
-                        offset += 1;
-                        linesToClear.push(linesToClear.shift()!);
-                    } else {
-                        linesOffset.set(lineIndex, offset);
-                    }
-                });
-        }
+        task.range
+            .slice()
+            .reverse()
+            .forEach((lineIndex) => {
+                if (lineIndex === linesToClear[0]) {
+                    // this line is an empty, cleared line. all lines above should fall with one more offset
+                    offset += 1;
+                    linesToClear.push(linesToClear.shift()!);
+                } else {
+                    linesOffset.set(lineIndex, offset);
+                }
+            });
         return linesOffset;
     }
 
     private scanLinesToClear(task: LineCheckTask): Array<number> {
         const linesToClear = [];
+        const WIDTH = BOARD_SIZE - 2 * WALL_SIZE;
         if (task.axis === LineCheckAxis.Row) {
             for (const row of task.range) {
                 // scan through each row
-                let canClear = true;
-                for (let col = 15; col < 25; col++) {
-                    if (this.board && this.board[row][col] === null) {
-                        canClear = false;
-                        break;
-                    }
-                }
+                const canClear = [...Array(WIDTH)]
+                    .map((_, i) => WALL_SIZE + i)
```

Comment:
Just for consistency with code above, I'd probably use `[...Array(WIDTH).keys()]` and then `.map((x) => //...`

Reply:
also this whole selected portion making the 15-25 array could be declared as a const above the if statement instead just `WIDTH`.